finality-aleph/src/sync/service.rs

```diff
@@ -730,54 +731,61 @@ where
     }
 
     /// Stay synchronized.
-    pub async fn run(mut self) {
+    pub async fn run(mut self) -> Result<Bottom, String> {
```

Comment:
Error types plx, here especially important since you get wrappers much more naturally than doing a `format!` in a `map_err`.

Reply:
done